ironfish/src/wallet/wallet.test.ts

```diff
@@ -1015,6 +1015,41 @@ describe('Wallet', () => {
       )
     })
 
+    it('should throw error if memo is too long', async () => {
+      const { node } = nodeTest
+
+      const accountA = await useAccountFixture(node.wallet, 'a')
+
+      const blockA1 = await useMinerBlockFixture(node.chain, undefined, accountA, node.wallet)
+      await expect(node.chain).toAddBlock(blockA1)
+
+      await node.wallet.updateHead()
+
+      const transaction = blockA1.minersFee
+
+      const transactionValue = await accountA.getTransaction(transaction.hash())
+      Assert.isNotUndefined(transactionValue)
+      Assert.isNotNull(transactionValue.sequence)
+
+      const rawTransaction = node.wallet.createTransaction({
+        account: accountA,
+        fee: 1n,
+        outputs: [
+          {
+            publicAddress: '0d804ea639b2547d1cd612682bf99f7cad7aad6d59fd5457f61272defcd4bf5b',
+            amount: 10n,
+            memo: Buffer.from('This is a string with 33 bytes!!!'),
+            assetId: Asset.nativeId(),
+          },
+        ],
+        expiration: 0,
+      })
+
+      await expect(rawTransaction).rejects.toThrow(
+        'The memo fields on the following outputs are more than the max of 32 bytes: 0',
+      )
+    })
```

Comment:
I think this whole test can be made a lot shorter. There's some cruft from copy pasting tests.

```suggestion
    it('should throw error if memo is too long', async () => {
      const account = await useAccountFixture(nodeTest.node.wallet, 'a')

      const promise = nodeTest.wallet.createTransaction({
        account: account,
        fee: 1n,
        outputs: [
          {
            publicAddress: account.publicAddress,
            amount: 1n,
            memo: Buffer.alloc(MEMO_LENGTH + 1),
            assetId: Asset.nativeId(),
          },
        ],
        expiration: 0,
      })

      await expect(promise).rejects.toThrow(MaxMemoLengthError)
    })
```